realm/realm-library/src/main/java/io/realm/ProxyUtils.java

```diff
@@ -0,0 +1,256 @@
+     * @param <E> type of the {@link RealmList}.
+     * @throws IOException if it fails to parse JSON stream.
+     */
+    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
+    static <E> RealmList<E> createRealmListWithJsonStream(Class<E> elementClass, JsonReader jsonReader) throws IOException {
+
+        if (jsonReader.peek() == null) {
+            jsonReader.skipValue();
+            return null;
+        }
+
+        jsonReader.beginArray();
+        RealmList realmList = new RealmList<E>();
+
+        if (elementClass == Boolean.class) {
+            while (jsonReader.hasNext()) {
+                if (jsonReader.peek() == JsonToken.NULL) {
+                    jsonReader.skipValue();
+                    realmList.add(null);
+                } else {
+                    realmList.add(jsonReader.nextBoolean());
+                }
+            }
+        } else if (elementClass == Float.class) {
+            while (jsonReader.hasNext()) {
+                if (jsonReader.peek() == JsonToken.NULL) {
+                    jsonReader.skipValue();
+                    realmList.add(null);
+                } else {
+                    realmList.add((float) jsonReader.nextDouble());
+                }
+            }
+        } else if (elementClass == Double.class) {
+            while (jsonReader.hasNext()) {
+                if (jsonReader.peek() == JsonToken.NULL) {
+                    jsonReader.skipValue();
+                    realmList.add(null);
+                } else {
+                    realmList.add(jsonReader.nextDouble());
+                }
+            }
+        } else if (elementClass == String.class) {
+            while (jsonReader.hasNext()) {
+                if (jsonReader.peek() == JsonToken.NULL) {
+                    jsonReader.skipValue();
+                    realmList.add(null);
+                } else {
+                    realmList.add(jsonReader.nextString());
+                }
+            }
+        } else if (elementClass == byte[].class) {
+            while (jsonReader.hasNext()) {
+                if (jsonReader.peek() == JsonToken.NULL) {
+                    jsonReader.skipValue();
+                    realmList.add(null);
+                } else {
+                    realmList.add(JsonUtils.stringToBytes(jsonReader.nextString()));
+                }
+            }
+        } else if (elementClass == Date.class) {
+            while (jsonReader.hasNext()) {
+                JsonToken token = jsonReader.peek();
+                if (token == JsonToken.NULL) {
+                    jsonReader.skipValue();
+                    realmList.add(null);
+                } else if (token == JsonToken.NUMBER) {
+                    realmList.add(new Date(jsonReader.nextLong()));
+                } else {
+                    realmList.add(JsonUtils.stringToDate(jsonReader.nextString()));
+                }
+            }
+        } else if (elementClass == Long.class) {
```

Comment:
Fold all these integers into one case?

Reply:
same ... needs cast ... otherwise it throws.